planemo/schema/test_file_schema.json

```diff
@@ -0,0 +1,3973 @@
+          "title": "Path",
+          "type": "string"
+        },
+        "expression": {
+          "description": "The regular expression to use.",
+          "title": "Expression",
+          "type": "string"
+        },
+        "negate": {
+          "anyOf": [
+            {
+              "type": "boolean"
+            },
+            {
+              "$ref": "#/$defs/PermissiveBooleanValue"
+            }
+          ],
+          "default": "false",
+          "description": "Negate the outcome of the assertion.",
+          "title": "Negate"
+        }
+      },
+      "required": [
+        "path",
+        "expression"
+      ],
+      "title": "AssertElementTextMatches",
+      "type": "object"
+    },
+    "AssertHasArchiveMember": {
+      "additionalProperties": false,
+      "description": "This tag allows to check if ``path`` is contained in a compressed file. The\npath is a regular expression that is matched against the full paths of the\nobjects in the compressed file (remember that \"matching\" means it is checked if\na prefix of the full path of an archive member is described by the regular\nexpression). Valid archive formats include ``.zip``, ``.tar``, and ``.tar.gz``.\nNote that.\n\ndepending on the archive creation method:\n- full paths of the members may be prefixed with ``./``\n- directories may be treated as empty files\n```xml\n&lt;has_archive_member path=\"./path/to/my-file.txt\"/&gt;\n```\nWith ``n`` and ``delta`` (or ``min`` and ``max``) assertions on the number of\narchive members matching ``path`` can be expressed. The following could be used,\ne.g., to assert an archive containing n&amp;plusmn;1 elements out of which at least\n4 need to have a ``txt`` extension.\n```xml\n&lt;has_archive_member path=\".*\" n=\"10\" delta=\"1\"/&gt;\n&lt;has_archive_member path=\".*\\.txt\" min=\"4\"/&gt;\n```\nIn addition the tag can contain additional assertions as child elements about\nthe first member in the archive matching the regular expression ``path``. For\ninstance\n```xml\n&lt;has_archive_member path=\".*/my-file.txt\"&gt;\n&lt;not_has_text text=\"EDK72998.1\"/&gt;\n&lt;/has_archive_member&gt;\n```\nIf the ``all`` attribute is set to ``true`` then all archive members are subject\nto the assertions. Note that, archive members matching the ``path`` are sorted\nalphabetically.\nThe ``negate`` attribute of the ``has_archive_member`` assertion only affects\nthe asserts on the presence and number of matching archive members, but not any\nsub-assertions (which can offer the ``negate`` attribute on their own).  The\ncheck if the file is an archive at all, which is also done by the function, is\nnot affected.\n$attribute_list::5",
+      "properties": {
+        "has_size": {
+          "items": {
+            "$ref": "#/$defs/AssertHasSize"
+          },
+          "title": "Has Size",
+          "type": "array"
+        },
```

Comment:
Instead of having all these assertions as properties we should have `"children"`, which contain a list of assertions, or. 

At least that's what the parser does : https://github.com/galaxyproject/galaxy/blob/9ddb6ad44afc59fadec39f5bdbf8e6b106774734/lib/galaxy/tool_util/parser/yaml.py#L291 .. or lets say is supposed to do (since the recursion is missing at the moment .. I guess this is easy to fix and I will try).

In hindsight it might have been better to just use `"asserts"` instead of `"children"` since its also the name of the top level list of asserts (or is this only in workflow tests).

Reply:
Just need to regenerate the schema, it's all derived from the xsd

Reply:
That was a comment on https://github.com/galaxyproject/planemo/pull/1417#discussion_r1516816920 ...

I am not sure I understand the proposition here, this doesn't seem necessary as you can simply use a list of assertions ?